src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,421 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.AspNet.OData.Formatter.Serialization;
+using Microsoft.AspNet.OData.Query.Expressions;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Query
+{
+    /// <summary>
+    /// Default implementation of SkipTokenHandler for the service. 
+    /// </summary>
+    public class DefaultSkipTokenHandler : SkipTokenHandler
+    {
+        private const char CommaDelimiter = ',';
+        private static char propertyDelimiter = '-';
+        internal static DefaultSkipTokenHandler Instance = new DefaultSkipTokenHandler();
+
+        /// <summary>
+        /// Constructor for DefaultSkipTokenHandler - Sets the Property Delimiter
+        /// </summary>
+        public DefaultSkipTokenHandler()
+        {
+        }
+
+        /// <summary>
+        /// Constructor for Unit testing purposes - Sets the Property Delimiter
+        /// </summary>
+        public DefaultSkipTokenHandler(char delimiter)
+            : this()
+        {
+            propertyDelimiter = delimiter;
+        }
+
+        /// <summary>
+        /// Returns the URI for NextPageLink
+        /// </summary>
+        /// <param name="baseUri">BaseUri for nextlink. It should be request URI for top level resource and navigationlink for nested resource.</param>
+        /// <param name="instance">Instance based on which SkipToken value will be generated.</param>
+        /// <param name="pageSize">Maximum number of records in the set of partial results for a resource.</param>
+        /// <param name="context">Serializer context</param>
+        /// <returns></returns>
+        public override Uri GenerateNextPageLink(Uri baseUri, int pageSize, Object instance, ODataSerializerContext context)
```

Comment:
>Uri baseUri, int pageSize [](start = 49, length = 25)

can these be obtained from the ODataSerializerContext?

Reply:
The top level request Uri and page size exist on the serializer context as part of internal request but the nested base Uri and page size are computed and do not exist on the serializer context. 

---
In reply to: [287397611](https://github.com/OData/WebApi/pull/1724#discussion_r287397611) [](ancestors = 287397611)